demux/demux_mkv.c

```diff
@@ -1862,6 +1869,45 @@ static int demux_mkv_open_sub(demuxer_t *demuxer, mkv_track_t *track)
     sh->codec->extradata = track->private_data;
     sh->codec->extradata_size = track->private_size;
 
+    if (!strcmp(sh->codec->codec, "arib_caption")) {
```

Comment:
Right after this if, you should probably check if `private_size` contains at least three bytes.